src/Resources/skeleton/doctrine/Entity.tpl.php

```diff
@@ -17,8 +17,8 @@ class <?= $class_name."\n" ?>
 {
     #[ORM\Id]
     #[ORM\GeneratedValue]
-    #[ORM\Column(type: 'integer')]
-    private int $id;
+    #[ORM\Column()]
+    private ?int $id = null;
```

Comment:
Side note - I think with the minimum versions we use for Doctrine now, we no longer need properties to be nullable. As I recall, this was _required_ in earlier versions otherwise doctrine would through an `exception` when initiating a class. E.g. 
```php
SomeEntity
{
    private int $id;

    public function getId(): int
    {
        return $this->id;
    }
    ...
}
```

This is no longer the case and the above works without issue. But it depends on which side of the coin we are looking at - for the purposes of Doctrine, `null` properties are not needed we are just adding unnecessary types (especially in terms of auto-generated data). In the terms of the codebase, nullable private properties _could_ make sense if a property is not init'd via the constructor. 

Personally, I lean more towards the doctrine train of thought - only make a property nullable if it is intended to be nullable.. Thoughts?

Reply:
I decided to make them nullable, not for Doctrine, but to avoid the PHP `must not be accessed before initialization` error that you'd get if you accessed the `$id` property (or any other property) before setting it / saving. I'm not positive it's the best approach, what I did is the most conservative: the code works exactly like the untyped version of the code. But, it does have "looser" types, so it's not *as* awesome.